src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -213,27 +213,33 @@ def get_tags(
 def get_context(group: Group) -> str:
     context_text = ""
 
-    context = group.issue_type.notification_config.context
-    context_dict = {}
-
-    for c in context:
-        if c in SUPPORTED_CONTEXT_DATA:
-            context_dict[c] = SUPPORTED_CONTEXT_DATA[c](group)
+    context = group.issue_type.notification_config.context.copy()
 
     # for errors, non-regression performance, and rage click issues
     # always show state and first seen
     # only show event count and user count if event count > 1 or state != new
 
-    event_count = context_dict.get("Events")
-    state = context_dict.get("State")
-    if (event_count and int(event_count) <= 1) or (state and state == "New"):
-        # filter out event count and users count
-        context_dict.pop("Events", None)
-        context_dict.pop("Users Affected", None)
+    state = None
+    event_count = ModuleNotFoundError
+    if "State" in context:
+        state = SUPPORTED_CONTEXT_DATA["State"](group)
+    if "Events" in context:
+        event_count = SUPPORTED_CONTEXT_DATA["Events"](group)
```

Comment:
could we have done this in place when we're copying the dict and trying to remove keys?